test/test_large_cohort.py

```diff
@@ -122,28 +133,271 @@ def create_config(
+        self, mocker: MockFixture, tmp_path: Path
+    ):
+        # Test that if conf.large_cohort['combiner']['intervals'] is empty
+        # then hl.vds.new_combiner is called with use_genome_default_intervals=True
+
+        # Creating a config and modyfing it in some way
+        conf = create_config(
+            tmp_path,
+            seq_type='genome',
+            gnomad_prefix=gnomad_prefix,
+            broad_prefix=broad_prefix,
+        )
+        conf.large_cohort['combiner']['intervals'] = []  # set to empty list
+
+        # Set config and patch cohort
+        set_config(
+            conf,
+            tmp_path / 'config.toml',
+            merge_with=[DEFAULT_CONFIG, LARGE_COHORT_CONFIG],
+        )
+
+        mocker.patch(
+            'cpg_workflows.inputs.create_cohort',
+            lambda: _mock_cohort(conf.workflow.dataset),
+        )
+
+        # Create a spy on hl.vds.new_combiner
+        spy_new_combiner = mocker.spy(hl.vds, 'new_combiner')
+
+        # Run the combiner function
+        start_query_context()
+        res_pref = tmp_path
+        vds_path = res_pref / 'v01.vds'
+        # FIXME: make this fast
+        combiner.run(out_vds_path=vds_path, tmp_prefix=res_pref / 'tmp')
+
+        # Check that hl.vds.new_combiner was called with the expected arguments
+        assert spy_new_combiner.call_args[1]['use_genome_default_intervals'] == True
+
+    def test_uses_default_exome_intervals_if_intervals_are_not_specified(
+        self, mocker: MockFixture, tmp_path: Path
+    ):
+        # Test that if conf.large_cohort['combiner']['intervals'] is empty
+        # then hl.vds.new_combiner is called with use_exome_default_intervals=True
+
+        # Hint: read about pytest_mock spy from the guide in the testing google document
+        pass
+
+    def test_fails_if_given_malformed_intervals(  # TODO: Finish creating test examples
+        self,
+        mocker: MockFixture,
+        tmp_path: Path,
+    ):
+        # See https://hail.is/docs/0.2/functions/genetics.html#hail.expr.functions.parse_locus_interval
+        # for valid formats
+
+        # Creating a config and modyfing it in some way
+        conf = create_config(
+            tmp_path,
+            seq_type='genome',
+            gnomad_prefix=gnomad_prefix,
+            broad_prefix=broad_prefix,
+        )
+
+        invalid = [
+            'this_shouldnt_work',  # Junk string
+            'CHR1:POS1-CHR1:100M',
+            'CHR2:POS2-CHR1:POS1',  # The start locus ("CHR2:POS2") must precede the end locus ("CHR1:POS1").
+            'CHR1:POS2-CHR1:POS1',  # The start position ("POS2") must be less than the end position ("POS1") within the same contig ("CHR1")
+            'CHR1:POS1-CHR1:START',  # The position "START" stands for 1, so it should not be used as an end position in the interval
+        ]
+
+        context_started = False
+
+        with open(tmp_path / 'intervals.txt'):  # test if can read intervals from txt
```

Comment:
This should be in a separate test named `test_can_load_intervals_from_a_text_file`